imagetest/test_suites/image_boot/image_boot_test.go

```diff
@@ -123,5 +197,49 @@ func TestBootTime(t *testing.T) {
 	if err != nil {
 		t.Fatalf("failed to convet start time %s", metadata)
 	}
-	t.Logf("image boot time is %d", time.Now().Second()-startTime)
+	t.Logf("image start time is %d", time.Now().Second()-startTime)
+}
+
+func TestBootTime(t *testing.T) {
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("couldn't get image from metadata")
+	}
+
+	MAX_THRESHOLD := getThresholdValue(image)
+
+	if MAX_THRESHOLD == 0 {
+		t.Fatalf("unrecognized image, no threshold value found")
+	}
+
+	var foundGuestAgentAndSshd bool
+
+	for i := 0; i < 120; i++ {
+		if err := lookForProcesses(); err == nil {
+			foundGuestAgentAndSshd = true
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	if !foundGuestAgentAndSshd {
+		t.Fatalf("Condition for guest agent and sshd process to start not reached within timeout")
+	}
+
+	// Reading the system uptime once both guest agent and sshd are found in the processes
+	uptimefile, err := os.ReadFile("/proc/uptime")
```

Comment:
uptimeContent or uptimeData seems to be a naming better reflecting its purpose for this variable (instead of uptimefile)